knative-operator/pkg/common/service_monitor.go

```diff
@@ -53,28 +54,31 @@ func SetupServerlessOperatorServiceMonitor(ctx context.Context, cfg *rest.Config
 	// Create Service object to expose the metrics port(s).
 	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
 	if err != nil {
-		log.Info("Could not create metrics Service", "error", err.Error())
+		return fmt.Errorf("failed to create metrics service: %w", err)
```

Comment:
@skonto did we ignore this error on purpose?

Reply:
The idea was that monitoring will be always there on OCP and we could move on if it was not setup (I used to have code that detected if monitoring is available like the coreos packaged does).

Reply:
Hm, sorry, not parsing that. Why'd we ignore an error during creation of the metrics service (a k8s service) and not all other errors then?

Reply:
Wait error is returned a few lines bellow.

Reply:
Well I didn't change anything about the flow to main, I think this purely makes the error handling a bit more "narrow" and ignores one somewhat common error that isn't really an error.

Reply:
Ok makes sense but afaik there was a discussion that we should never hit this.